src/charm.py

```diff
@@ -33,83 +31,59 @@ def __init__(self, *args):
             args: Arguments passed to the CharmBase parent constructor.
         """
         super().__init__(*args)
-        self.framework.observe(self.on.httpbin_pebble_ready, self._on_httpbin_pebble_ready)
+        try:
+            self._charm_state = CharmState.from_charm(charm=self)
+        except CharmConfigInvalidError as exc:
+            self.model.unit.status = ops.BlockedStatus(exc.msg)
```

Comment:
I'm not a too big fan of having exc.msg as the charm status, would "Invalid Charm Status, please see juju debug-logs" be better for longer error messages? 
On the other hand, having explicit messages can be a lot easier to see at first glance though.

Reply:
That was the idea. Under this scenario, we'll be able to see the invalid fields directly in the status